src/MVVM/Factories/ViewModelFactory.cs

```diff
@@ -142,7 +141,133 @@ internal static INavigationService TryResolveNavigationServiceInstance()
+                        $"The type {viewModelType.Name} must implement interface type {interfaceType.Name}");
+                }
+
+                var instance = iocContainer.Resolve(viewModelType) as IViewModel;
+
+                if (initialize)
+                    TaskHelper.RunSync(instance.InitializeAsync);
+
+                return instance;
+            }
+            catch (Exception innerException)
+            {
+                string errorMessage = $"ViewModel of type {interfaceType.Name} cannot be resolved --- is it registered?";
+
+                log?.Critical(
+                    errorMessage,
+                    innerException);
+
+                throw new TypeLoadException(
+                    errorMessage,
+                    innerException);
+            }
+        }   
+        
+
+        /// <summary>
+        /// Creates a new ViewModel instance and optionally invokes the <see cref="IViewModel.InitializeAsync"/> method.
+        /// </summary>
+        /// <param name="viewModelType">Type of the ViewModel to create</param>
+        /// <param name="initialize">Specifies if <see cref="IViewModel.InitializeAsync"/> should get called after instance creation.</param>
+        /// <returns>A ViewModel instance of the given type</returns>
+        public static async Task<IViewModel> ResolveAsync(
+            Type viewModelType,
+            bool initialize = true)
+        {
+            var interfaceType = typeof(IViewModel);
+
+            try
+            {
+                if (!interfaceType.IsAssignableFrom(viewModelType))
+                {
+                    throw new InvalidOperationException(
+                        $"The type {viewModelType.Name} must implement interface type {interfaceType.Name}");
+                }
+
+                var instance = iocContainer.Resolve(viewModelType) as IViewModel;
+
+                if (initialize)
+                    await instance.InitializeAsync();
+
+                return instance;
+            }
+            catch (Exception innerException)
+            {
+                string errorMessage = $"ViewModel of type {interfaceType.Name} cannot be resolved --- is it registered?";
+
+                log?.Critical(
+                    errorMessage,
+                    innerException);
+
+                throw new TypeLoadException(
+                    errorMessage,
+                    innerException);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new ViewModel instance and invokes the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method.
+        /// </summary>
+        /// <typeparam name="TModel">Type of the parameter that will be passed in the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method</typeparam>
+        /// <param name="viewModelType"></param>
+        /// <param name="model">Parameter that will be passed in the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method</param>
```

Comment:
Codacy found an issue: [Split this 140 characters long line (which is greater than 130 authorized).](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5767060)